kernel/Utilities/SparsityEstimator.h

```diff
@@ -42,26 +43,11 @@ class SparsityEstimator {
                                  std::vector<int>& nonZeroPerRowNonOwned,
                                  bool includeFaceCoupling = true) const;
 
-   private:
-    /// Workspace variables for the computation
-    struct Workspace;
-
-    /// Add the DoFs that have support on an Element to the workspace.
-    void addElementDoFs(const Base::Element* element,
-                        Workspace& workspace) const;
-
-    /// Write the DoF count for an Element, Face, etc. to the sparsity estimate
-    /// vector \tparam GEOM The Element, Face, etc. type \param geom The
-    /// Element, Face, etc. which supports the basis functions for which to
-    /// write the DoF count. \param workspace The workspace with DoF counts
-    /// \param nonZeroPerRowOwned see computeSparsityEstimate(std::vector<int>&,
-    /// std::vector<int>&) \param nonZeroPerRowNonOwned see
-    /// computeSparsityEstimate(std::vector<int>&, std::vector<int>&)
-    template <typename GEOM>
-    void writeDoFCount(const GEOM* geom, const Workspace& workspace,
-                       std::vector<int>& nonZeroPerRowOwned,
-                       std::vector<int>& nonZeroPerRowNonOwned) const;
+    void computeSparsityEstimate(std::vector<int>& nonZeroPerRowOwned,
+                                 std::vector<int>& nonZeroPerRowNonOwned,
```

Comment:
I think it is nicer if it returned a std::pair or a struct instead of having two non-const reference arguments.